### PR TITLE
[zh] docs: fix incorrect display on rust-based-ztunnel

### PR DESCRIPTION
**Pull request by charles-chenzz**
Please provide a description for what this PR is for.
this should fix the incorrect display on a recently posted blog
before: 
image

@windsonsea PTAL
And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Ambient
- [X] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by charles-chenzz** [comment]
the fix didn't work, still looking into it

File: content/zh/blog/2023/rust-based-ztunnel/index.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ Ztunnel 确保流量高效、安全地传输到 waypoint proxy，从而实现 HT
     caption="ztunnel architecture"
     >}}
 
-## 为什么不重用 Envoy？{#why-not-reuse-envoy?}
+## 为什么不重用 Envoy?{#why-not-reuse-envoy?}
```
Comment (windsonsea) [inline content/zh/blog/2023/rust-based-ztunnel/index.md:32]:
```suggestion
## 为什么不重用 Envoy? {#why-not-reuse-envoy}
```
Do not add any symbol in anchor

Comment (windsonsea) [reply content/zh/blog/2023/rust-based-ztunnel/index.md]: That's all right. It's not required to squash commits in istio.io.

**Review by windsonsea — APPROVED** [review]
Thanks. LGTM

**Pull request merged**